### PR TITLE
#187 Add Conan recipe

### LINKED ISSUE #187: Conan Package Support
Hello,

Do you know about [Conan](https://github.com/conan-io)?

Conan is modern dependency and package manager for C/C++. And will be great if your library will be available via package manager for other developers.

Now we have famous C++ projects, including Google Flatbuffers. We could support you writing a recipe and publishing your packages on Bintray.

If you have any questions, just ask :-)

### PR DESCRIPTION
**Pull request by uilianries**
Hi!

This PR is related to #187 which provides Conan package support for Google Abseil.

The recipe included on this PR contains some metadata, as author, project url, license, and steps needed to create the package.
As [cctz](https://github.com/google/cctz) it's listed as requirement for Abseil and will be downloaded automatically when building Abseil using Conan.

How to build Abseil using Conan?

    $ conan create . abseil/latest@abseil/stable

Conan [create](https://docs.conan.io/en/latest/reference/commands/creator/create.html) will execute all steps, including download cctz and create a binary package for Abseil.

The PR https://github.com/abseil/abseil-cpp/pull/197 is related to CI support, which is awesome, because we could include Conan on CI build steps to distribute Abseil as pre-built binaries. Of course, this should be done in a next PR.

Regards!

closes #187 

I'm adding some Conan experts to review this PR:
/cc @danimtb @memsharded

### CODE REVIEW

**Comment by rongjiecomputer** [comment]
> As cctz it's listed as requirement for Abseil and will be downloaded automatically when building Abseil using Conan.

Abseil no longer require downloading CCTZ. CCTZ is already part of Abseil under `absl/time/internal`.

**Comment by uilianries** [comment]
Great! I'm going to remove it, thanks for reviewing

**Comment by uilianries** [comment]
Done, CCTZ removed.

**Review by danimtb — APPROVED** [review]
LGTM. Having support for Conan in this library will be great! Thanks @uilianries

**Review by Croydon — APPROVED** [review]
You might want to add `topics`, LGTM otherwise

**Comment by uilianries** [comment]
well observed, I'm gonna update this.

File: conanfile.py
Comment (ahedberg) [inline conanfile.py]: I'm pretty sure I didn't write this?

Comment (uilianries) [reply conanfile.py]: `author` is a reference of who is responsible for the recipe. we could change of course, do you prefer use other name or team name?

Comment (ahedberg) [reply conanfile.py]: Let's put the Abseil mailing list (abseil-io@googlegroups.com) here instead.

File: conanfile.py
Comment (ahedberg) [inline conanfile.py:14]: There was discussion on the associated issue about what to do with the "version" tag. I don't see it here--will folks using Abseil through Conan get an LTS, or a particular commit? Will they have to "sync" or "update" every time we push a new commit to GitHub?

File: conanfile.py
Comment (ahedberg) [inline conanfile.py:1]: We should add a comment about this being supported on a best-effort basis. We're okay with adding this to our repository, but since we don't use Conan, we won't know if it stops working, and we may need community members to help us debug problems that arise.

**Review by ahedberg — CHANGES_REQUESTED** [review]

File: conanfile.py
Comment (uilianries) [reply conanfile.py:1]: No problem, you could invoke any Conan member to help you, including me.

File: conanfile.py
Comment (uilianries) [reply conanfile.py:14]: excellent question. This first commit is only related to provide Conan recipe itself, but don't build it using any CI service. As `version` is not listed in this recipe, you will need to specific the version tag when creating a new package, for example:

    conan create . abseil/<version>@abseil/stable

*version* could be your current date, commit, or even "latest". We would prefer using date e.g. *abseil/20181220@abseil/stable* because is easier to sort.

We could support LTS version too, but is outdated, when compared to your latest version, so you will be able to provide as many as you want.

I'll provide a new PR in the future, containing all changes needed to build 
Conan package in your CI, but as abseil-cpp doesn't have CI support yet, I'll need to wait for https://github.com/abseil/abseil-cpp/pull/197.

File: conanfile.py
Comment (uilianries) [reply conanfile.py]: done

File: conanfile.py
Comment (ahedberg) [reply conanfile.py:14]: Got it, thanks. We can hold off on deciding what to use for the version once we're ready to add it.

**Review by ahedberg — APPROVED** [review]

**Comment by ahedberg** [comment]
This will be merged by @mbxx later today as part of our daily release process.

**Comment by uilianries** [comment]
Thank you!

**Pull request merged**
